src/compiler/compiler.rs

```diff
@@ -355,6 +360,20 @@ impl Compiler {
                 storage.get(&key)
             };
 
+            // Wait at most a minute for the cache to respond before we forge
+            // ahead ourselves with a compilation.
+            let timeout = Duration::new(60, 0);
```

Comment:
As a followup, it might be nice to have a way to configure this so we can tweak it in practice.